features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapSinkModule.java

```diff
@@ -82,25 +86,25 @@ public int getCompletionIntervalMs() {
 
             @Override
             public Object key(TrapInformationWrapper message) {
-                return message.getTrapInformation().getTrapAddress();
+                if (message.getTrapInformation() != null) {
+                    return message.getTrapInformation().getTrapAddress();
+                } else {
+                    return message.getTrapDTO().getTrapAddress();
```

Comment:
Can we maybe just call `message.getTrapAddress()` instead, and hide this logic in the `TrapInformationWrapper` itself?